src/test/java/org/mockito/internal/stubbing/defaultanswers/ReturnsEmptyValuesTest.java

```diff
@@ -108,7 +108,64 @@ public void should_return_empty_optional() throws Exception {
     }
 
     @Test
-    public void should_return_empty_stream() throws Exception {
+    public void should_return_empty_OptionalDouble() throws Exception {
+        Class<?> streamType = getClassOrSkipTest("java.util.stream.DoubleStream");
+
+        //given
+        Object stream = mock(streamType);
+        Object optional = streamType.getMethod("findAny").invoke(stream);
+        assertNotNull(optional);
+        assertFalse((Boolean) Class.forName("java.util.OptionalDouble").getMethod("isPresent").invoke(optional));
+
+        Invocation findAny = this.getLastInvocation();
+
+        //when
+        Object result = values.answer(findAny);
+
+        //then
+        assertEquals(optional, result);
+    }
+
+    @Test
+    public void should_return_empty_OptionalInt() throws Exception {
```

Comment:
It's a poor man's parameterized tests, but just for that case it is probably enough.

Btw, I haven't used `Theories`, but `Parameterized` from JUnit is very verbose and limited. To define one parameterized test you need to sacrifice the whole class (you need to use fields and a constructor). There's no way to create more than one parameterized test in the same test class (which very often is very handy). JUnitParams is much more flexible (no to mentioned Spock brings it to the another level :) ).